src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
-// write code here
+const list = document.querySelectorAll('ul li');
+
+function sortList() {
+  const listItems = [...list];
+
+  const numeric = () => {
+    for (const item of listItems) {
+      item.dataset.salary = Number(
+        item.dataset.salary.replace(/[^0-9.-]+/g, ''),
+      );
+    }
+  };
+
+  numeric();
+
+  listItems.sort((a, b) => b.dataset.salary - a.dataset.salary);
```

Comment:
The sorting function assumes that all 'li' elements have a 'data-salary' attribute. If any element lacks this attribute, it could lead to unexpected behavior. Consider adding a check to ensure the attribute exists before attempting to access it.